File: src/beginner_tutorials/src/Perez_listener.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"
#include "std_msgs/String.h"

void velCallback(const geometry_msgs::Twist& msg)
{
  if(msg.linear.x > 0.0)
    ROS_INFO("GOING FORWARD");
  else if(msg.linear.x < 0.0)
    ROS_INFO("GOING BAKWARD");

  if(msg.angular.z > 0.0)
    ROS_INFO("TURNING LEFT");
  else if(msg.angular.z < 0.0)
    ROS_INFO("TURNING RIGHT");

}

int main(int argc, char **argv){
  ros::init(argc, argv, "listener");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe("cmd_vel", 1000, velCallback);
  ros::spin();

  return 0;
}

File: src/beginner_tutorials/scripts/Perez_listener.py
#!/usr/bin/env python
import rospy
from geometry_msgs.msg import Twist

def callback(msg):
    if msg.linear.x > 0.0:
        rospy.loginfo("GOING FORWARD")
    elif msg.linear.x < 0.0:
        rospy.loginfo("GOING BAKWARD")

    if msg.angular.z > 0.0:
        rospy.loginfo("TURNING LEFT")
    elif msg.angular.z < 0.0:
        rospy.loginfo("TURNING RIGHT")

def listener():

    rospy.init_node('Perez_listener_py', anonymous=True)

    rospy.Subscriber('cmd_vel', Twist, callback)

    rospy.spin()

if __name__ == '__main__':
    listener()
